### PR TITLE
move LayoutSpecification types to DynamoCore with type forward

### PR DESCRIPTION
**Pull request by mjkkirschner**
@SHKnudsen @aparajit-pratap @QilongTang
PTAL - 
The purpose of this PR is remove the DynamoCore.Wpf dependency from the NodeMarkdown docs generator CLI.

this PR uses type forwarding to move the 5 layout spec types from dynamoCore to DynamoCoreWPF:
https://docs.microsoft.com/en-us/dotnet/standard/assembly/type-forwarding

Theres no reason they need to be in DynamoCore.Wpf anyway, they have no deps on WPF.

this should allow old callers to continue working without recompiling - for those clients who do try to recompile on a newer Dynamo, all they should need to do is add `Dynamo.Interfaces` to their list of using statements.

### CODE REVIEW

File: src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 using System.Reflection;
 using System.Text;
 using System.Text.RegularExpressions;
-using Dynamo.Wpf.Interfaces;
+using Dynamo.Interfaces;
```
Comment (aparajit-pratap) [inline src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs:9]: Does the code not compile any longer without these additional `using`s or are these required at runtime. Even though the type forwarding includes the complete namespace, is this still required?

File: src/Tools/NodeDocumentationMarkdownGenerator/NodeDocumentationMarkdownGenerator.csproj [hunk 2]
```diff
@@ -132,11 +132,6 @@
     </EmbeddedResource>
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\..\DynamoCoreWpf\DynamoCoreWpf.csproj">
-      <Project>{51bb6014-43f7-4f31-b8d3-e3c37ebedaf4}</Project>
-      <Name>DynamoCoreWpf</Name>
-      <Private>False</Private>
-    </ProjectReference>
```
Comment (aparajit-pratap) [inline src/Tools/NodeDocumentationMarkdownGenerator/NodeDocumentationMarkdownGenerator.csproj:139]: 👍

**Review by aparajit-pratap — APPROVED** [review]
just one Q

**Review by SHKnudsen — APPROVED** [review]

**Comment by mjkkirschner** [comment]
So the code won’t compile without these usings - because I changed the namespace the type belonged to. But it’s my understanding from the docs on the type forward attribute that previously compiled clients using these types will be able to find them without issue and won’t need to recompile. For clients that do recompile they’ll need to do what I did here and add the usings.

Or - I could keep the namespace the same to avoid that I believe, do you have a preference?

MK

**Comment by aparajit-pratap** [comment]
> Or - I could keep the namespace the same to avoid that I believe, do you have a preference?

@mjkkirschner no, this is fine, LGTM!

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**